pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -1006,9 +1097,131 @@ func (r *ReconcileClusterPool) createCloudBuilder(pool *hivev1.ClusterPool, logg
 		cloudBuilder.Region = platform.Azure.Region
 		cloudBuilder.CloudName = platform.Azure.CloudName
 		return cloudBuilder, nil
-	// TODO: OpenStack, VMware, and Ovirt.
+	case platform.OpenStack != nil:
+		credsSecret, err := r.getCredentialsSecret(pool, platform.OpenStack.CredentialsSecretRef.Name, logger)
+		if err != nil {
+			return nil, err
+		}
+		cloudBuilder := clusterresource.NewOpenStackCloudBuilderFromSecret(credsSecret)
+		cloudBuilder.Cloud = platform.OpenStack.Cloud
```

Comment:
This is also set in `NewOpenStackCloudBuilderFromSecret`. I think we can remove it from there as it is not fetched from secret

Reply:
Agree, and because it is a required field